x-pack/filebeat/module/cisco/shared/ingest/asa-ftd-pipeline.yml

```diff
@@ -366,7 +368,7 @@ processors:
       field: "message"
       description: "304001"
       patterns:
-      - "%{IP:source.address} %{DATA} (%{NOTSPACE}@)?%{IP:destination.address}:%{GREEDYDATA:url.original}"
+      - "%{IP:source.address}(\\((%{DATA:source.user.domain}\\\\)?%{USERNAME:source.user.name}\\))? %{DATA} (%{NOTSPACE}@)?%{IP:destination.address}:%{GREEDYDATA:url.original}"
```

Comment:
Some of our usernames are Active Directory UPNs where they are of the format USER@DOMAIN.TLD. These can fail grok matching because the USERNAME pattern does not include @ as a matching character.

user@domain.tld@192.168.0.1(LOCAL\user@domain.tld) Accessed URL 82.0.0.1:http://www.someurl.tld/index.html

Other patterns that match this field use NOTSPACE or GREEDYDATA and ignore the domain portion since its always LOCAL and isn't representative of the users actual domain.

Reply:
So is LOCAL not the domain?

Reply:
There are also a few message-ids where the various *.user.name fields include the surrounding brackets.
Message-id 302014 and some others are examples.

Reply:
> So is LOCAL not the domain?

Documentation suggests it is the domain but in our environment, even with domain users, it is always LOCAL. There maybe some configuration that controls that.

Reply:
The other patterns ignore the domain and include it in the *.user.name. Is it worth bringing the others inline with this change too?

Reply:
I think my latest commit addresses these issues.